mnist-dp/assets/algo_cnn_dp/algo.py

```diff
@@ -0,0 +1,116 @@
+import substratools as tools
+import os
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+
+tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.ERROR)
+
+from tensorflow_privacy.privacy.analysis import compute_dp_sgd_privacy
+from tensorflow_privacy.privacy.optimizers.dp_optimizer import DPGradientDescentGaussianOptimizer
+
+# input image dimensions
+img_rows, img_cols = 28, 28
+input_shape = (img_rows, img_cols, 1)
+num_classes = 10
+
+#model hyperparameter values
+epochs = 1
+batch_size = 250
+
+# DP-SGD privacy-specific hyperparameters
+l2_norm_clip = 1.5
+noise_multiplier = 1.3
+num_microbatches = 250
```

Comment:
Maybe add one line description (or link) to these hyper-parameters.

Reply:
Good idea!